presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -326,16 +325,20 @@ public ConnectorTableMetadata getTableMetadata()
             @Override
             public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
             {
-                Iterable<List<Object>> records = () ->
-                        stream(partitionManager.getPartitions(metastore, sourceTableHandle, alwaysTrue()).getPartitions())
-                                .map(hivePartition -> {
-                                    List<Object> values = new ArrayList<>();
-                                    for (int i = 0; i < partitionColumns.size(); ++i) {
-                                        values.add(hivePartition.getKeys().get(fieldIdToColumnHandle.get(i)).getValue());
-                                    }
-                                    return values;
-                                })
-                                .iterator();
+                Iterable<List<Object>> records = () -> {
+                    TupleDomain<ColumnHandle> targetTupleDomain = constraint.transform(fieldIdToColumnHandle::get);
```

Comment:
I think you can pull these outside the lambda, since they only need to happen once (seems cleaner even though in practice this is only called once)